modules/simple_forms_api/app/services/simple_forms_api/prefill_data_service.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module SimpleFormsApi
+  class PrefillDataService
+    attr_reader :prefill_data, :form_data, :form_id
+
+    def initialize(prefill_data:, form_data:, form_id:)
+      @prefill_data = JSON.parse(prefill_data)
+      @form_data = form_data
+      @form_id = form_id
+    end
+
+    def check_for_changes
+      changed_fields = []
+      changed_fields << 'first_name' if prefill_data.dig('full_name', 'first') != form_data.dig('full_name', 'first')
```

Comment:
Do we want to add case insensitivity here?

Reply:
I don't think so. This check is just for information at the moment, we're not acting on it directly (and not necessarily intending to). So it might be interesting to surface when people change capitalization of their names.